### PR TITLE
Add 'link status' indication for network adapters and other small things

### PR DESCRIPTION
**Pull request by twvd**
Changes in this PR:
* Read the link speed and duplex for all supported network adapters and display them in the applet and SystemMonitor.
* Fix for RTL8139-driver not working because link status is not initialized.
* Friendlier representation of a NIC that has no IPv4-address.

How things look when the link is up:
image
image

How things look when the link is down:
image
image

I didn't test the RTL8168 driver as qemu cannot emulate that device

### CODE REVIEW

File: Userland/Applets/Network/main.cpp [hunk 1]
```diff
@@ -134,7 +137,12 @@ class NetworkWidget final : public GUI::ImageWidget {
             if (!adapter_info.is_empty())
                 adapter_info.append('\n');
 
-            adapter_info.appendff("{}: {}", ifname, ip_address);
+            adapter_info.appendff("{}: {} ", ifname, ip_address);
+            if (!link_up)
+                adapter_info.appendff("(down)");
+            else
+                adapter_info.appendff("({} Mb/s, {}-duplex)", link_speed,
+                    link_fd ? "full" : "half");
```
Comment (alimpfard) [inline Userland/Applets/Network/main.cpp:145]: Please put multiline branches in braces:
```suggestion
            if (!link_up) {
                adapter_info.appendff("(down)");
            } else {
                adapter_info.appendff("({} Mb/s, {}-duplex)", link_speed,
                    link_fd ? "full" : "half");
            }
```

Comment (alimpfard) [inline Userland/Applets/Network/main.cpp:145]: I'm not entirely sure if information about whether the connection is full or half-duplex is useful enough to put on a tooltip tbh

File: Userland/Applications/SystemMonitor/NetworkStatisticsWidget.cpp [hunk 2]
```diff
@@ -48,6 +48,14 @@ NetworkStatisticsWidget::NetworkStatisticsWidget()
         net_adapters_fields.empend("name", "Name", Gfx::TextAlignment::CenterLeft);
         net_adapters_fields.empend("class_name", "Class", Gfx::TextAlignment::CenterLeft);
         net_adapters_fields.empend("mac_address", "MAC", Gfx::TextAlignment::CenterLeft);
+        net_adapters_fields.empend("Link status", Gfx::TextAlignment::CenterLeft,
+            [this](JsonObject const& object) -> String {
+                if (!object.get("link_up").as_bool())
+                    return "Down";
+                else
```
Comment (alimpfard) [inline Userland/Applications/SystemMonitor/NetworkStatisticsWidget.cpp:55]: No `else` after `return`.

File: Kernel/Net/RTL8139NetworkAdapter.cpp [hunk 3]
```diff
@@ -260,6 +260,9 @@ void RTL8139NetworkAdapter::reset()
     // choose irqs, then clear any pending
     out16(REG_IMR, INT_RXOK | INT_RXERR | INT_TXOK | INT_TXERR | INT_RX_BUFFER_OVERFLOW | INT_LINK_CHANGE | INT_RX_FIFO_OVERFLOW | INT_LENGTH_CHANGE | INT_SYSTEM_ERROR);
     out16(REG_ISR, 0xffff);
+
+    // fill initial link up status
```
Comment (alimpfard) [inline Kernel/Net/RTL8139NetworkAdapter.cpp:264]: How about:
```suggestion
    // Set initial link up status.
```

File: Userland/Applets/Network/main.cpp (hunk 1 above)
Comment (twvd) [reply Userland/Applets/Network/main.cpp]: I removed it

Comment (twvd) [reply Userland/Applets/Network/main.cpp]: Removed duplex so now it all fits on one line :)

**Comment by twvd** [comment]
Hi @alimpfard , thanks for your review! I have addressed your comments, ready for another review.

**Review by alimpfard — APPROVED** [review]
LGTM

**Pull request merged**